### PR TITLE
Simple regex replace.

### PR DESCRIPTION
**Pull request by nightblade9**
Code review please @bsinky .

I was initially thinking of extending `Match` to return the start/end position (Haxe provides a `matchedLeft` and `matchedRight` string), but that ended up melting my brain, so I went with a much simpler approach here.

This class is a bit strange because it creates/uses/disposes an `EReg` instance at runtime (doesn't keep it for long-lived operations). Not sure if that will pan out advantageously or not.

### CODE REVIEW

File: src/haxesharp/text/Regex.hx [hunk 1]
```diff
@@ -75,7 +75,27 @@ class Regex
...
+        var groups = Regex.matches(input, this.regex, this.options);        
         return new Match(groups.length > 0, groups);
     }
+
+    /**
+    Finds all instances where this regex matches the input string, and
+    replaces those with the replacement string. Does not (yet) support
+    groups or named groups.
+    **/
+    public function replace(input:String, replacement:String):String
```
Comment (bsinky) [inline src/haxesharp/text/Regex.hx:87]: `replacement` variable is unused, which means something like `.replace(myRegex, "test")` is just removing any matches of `myRegex` without substituting in the `replacement` string.

File: test/haxesharp/text/RegexTest.hx [hunk 2]
```diff
@@ -68,4 +68,22 @@ class RegexTest
...
+        Assert.that(input != "test string");
+    }
+
+    @Test
+    public function replaceReturnsInputIfNothingMatches()
+    {
+        var input = "test string";
+        var output = new Regex("\\d+", "g").replace(input, "");
+        Assert.that(output, Is.equalTo(input));
+    }
```
Comment (bsinky) [inline test/haxesharp/text/RegexTest.hx:88]: We should add a test case for non-empty `replacement` values to ensure that these also produce the expected results.

File: src/haxesharp/text/Regex.hx (hunk 1 above)
Comment (nightblade9) [reply src/haxesharp/text/Regex.hx:87]: This is the main use case I was trying to replace (remove comments in a JSON file). I totally missed this. I'll add a test case too. Good catch!

File: test/haxesharp/text/RegexTest.hx [hunk 3]
```diff
@@ -68,4 +68,22 @@ class RegexTest
...
     }
+
+    @Test
+    public function replaceReplacesAllInstancesIfMultipleExist()
+    {
+        var input = "44test1 str22i2ng3";
+        var output = new Regex("\\d+", "g").replace(input, "");
+        Assert.that(output, Is.equalTo("test string"));
+        // Didn't butcher the input, did we?
+        Assert.that(input != "test string");
```
Comment (nightblade9) [inline test/haxesharp/text/RegexTest.hx]: This actually needs to be negated. We want to test that the input string is still whatever it was before.

**Comment by nightblade9** [comment]
I think this is ready for review again.

**Review by bsinky — APPROVED** [review]

**Comment by bsinky** [comment]
Everything looks good to me. 👍
